### DISCUSSION TITLE
可否部署到 阿里云函数计算 或者 腾讯云Serverless 上

### DISCUSSION THREAD

**Discussion by yb** [chanify/chanify#68] [Q&A]
我个人一直觉得将这类微服务部署到 函数计算、Serverless 是非常合适的，虽然它们默认不带数据库、不支持数据持久化，但也都有相应的方案（例如 sqlite 文件可以放到持久存储，例如有额外的 mysql 服务），而且 chanify 自己也支持 serverless 模式。

所以我尝试了将 chanify 部署到阿里云的函数计算，但函数计算的弯弯绕实在太多，没有成功。

不知道有没有人部署成功的，可以给点思路？

**Comment by wizjin** [comment]
你好，这个需要集成他们的 SDK 才可以工作。

大致思路是利用 HTTP 触发来调用 Chanify 的代码。

但是目前 `阿里云函数计算` 或者 `腾讯云 Serverless` 都暂时都不支持 Golang。

**Reply by yb** [reply]
谢谢大佬回复。
现在他们已经支持自定义容器，感觉有点类似 docker，好像也可以直接用 docker 的镜像，但 docker 和自定义容器我都不是很了解，所以看着有点懵。
这个 go-demo 可能会有帮助：https://github.com/awesome-fc/fc-custom-demo/tree/master/go-demo
补充一个链接：https://github.com/awesome-fc/custom-container-docs/tree/master/go-gin

**Comment by yb** [comment]
更新一下昨天尝试的结果：
阿里云函数计算 不需要用 `docker` 和 `custom-container`，只需要在 `template.yml` 中配置 `runtime: custom`，然后在 `bootstrap` 文杰中启动编译好的 chanify 就行。就算是冷启动，首次请求也可以在 2s 内完成，可以接受。

**Comment by wizjin** [comment]
@yb 十分感谢分享
